### PR TITLE
Archivelistwidget doxygen

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
This contains PR #130 so that the `\ref RestoreDialog` link works.

File: src/widgets/archivelistwidget.h [hunk 1]
```diff
@@ -7,30 +7,53 @@
...
 public:
+    //! Constructor.
     explicit ArchiveListWidget(QWidget *parent = nullptr);
     ~ArchiveListWidget();
 
 public slots:
+    //! Adds archives to the displayed list, creating an
+    //! ArchiveListWidgetItem for every archive.
     void addArchives(QList<ArchivePtr> archives);
+    //! Displays detailed information about a single archive.
```
Comment (shinnok) [inline src/widgets/archivelistwidget.h]: This only sets the current selection in the list view.

File: src/widgets/archivelistwidget.h [hunk 2]
```diff
@@ -7,30 +7,53 @@
...
     void setSelectedArchive(ArchivePtr archive);
+    //! Disable the ArchiveListWidgetItems corresponding to the
+    //! specified archives.
     void disableArchives(QList<ArchivePtr> archives);
```
Comment (shinnok) [inline src/widgets/archivelistwidget.h]: You can use just "items" here instead of ArchiveListWidgetItems.

File: src/widgets/archivelistwidget.h [hunk 3]
```diff
@@ -7,30 +7,53 @@
...
     Q_OBJECT
 
 public:
+    //! Constructor.
     explicit ArchiveListWidget(QWidget *parent = nullptr);
     ~ArchiveListWidget();
 
 public slots:
+    //! Adds archives to the displayed list, creating an
+    //! ArchiveListWidgetItem for every archive.
```
Comment (shinnok) [inline src/widgets/archivelistwidget.h]: See commit 55f78965.

**Comment by gperciva** [comment]
Updated.  I'm not certain that I understand the `addArchives()` -> `setArchives()` change.  Looking at `archivelistwidget.cpp`, it seems that `setArchives()` sorts the list of archives, then creates a new ArchiveListWidgetItem for every archive.

**Comment by gperciva** [comment]
Ignore the previous comment about `setArchives()`; I didn't realize what the `clear()` was doing.  Updated PR accordingly.

**Pull request merged**